### PR TITLE
(maint) Readability udpates for pa_matchbatch.sh

### PR DESCRIPTION
**Pull request by caseywilliams**
These are updates that clarify some things for puppet-agent release docs; See notes inline.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,37 +75,44 @@ Usage: ruby ticketmatch.rb [options]
     -h, --help                       this message
 ```
 
-# Batch processing
+# Batch processing for puppet-agent and its components
```
Comment (caseywilliams) [inline README.md:78]: (since this only applies to puppet-agent)

File: README.md [hunk 2]
```diff
@@ -75,37 +75,44 @@ Usage: ruby ticketmatch.rb [options]
...
-`pa_matchbatch.sh`. It requires one argument: the revision (SHA or reference) of `puppet-agent` you
-care about. It will process `puppet-agent` and all its components that are tagged with a SHA.
 
-Component directories will be cloned from the path provided in their `component.json` file and the
-appropriate branch set. If a given component already exists, a Git fetch will be performed to get
-things up to date (see environment variables, below, for additional options).
+The `pa_matchbatch.sh` script clones a revision of puppet-agent and all of its components, then runs
+ticketmatch on everything; This is intended for use during puppet-agent releases.
 
-## How to run pa_matchbatch (one way)
```
Comment (caseywilliams) [inline README.md:88]: I didn't understand why there were two ways described here, so I replaced this section with one way

File: README.md [hunk 3]
```diff
@@ -75,37 +75,44 @@ Usage: ruby ticketmatch.rb [options]
...
 
-```path/pa_matchbatch.sh <the_revision>```
+*`pa_matchbatch.sh` clones puppet-agent and its components to the current working directory by default.*
+You can override the target directory by setting the `$WORKSPACE` environment variable to some other
+directory. You can run `pa_matchbatch.sh` any number of times on the same workspace; The repos will
+be fetched each time to check for new updates.
 
-## How to run pa_matchbatch (another way)
+*`pa_matchbatch.sh` assumes that only the Z versions of puppet-agent and its components will change
+during a release.* You will need to override this assumption for:
```
Comment (caseywilliams) [inline README.md:99]: The description of why versions might be incorrect was incomplete (the note about Z versions was the reason I started these edits), and this is quite important, so I moved it upward

File: pa_matchbatch.sh [hunk 4]
```diff
@@ -2,15 +2,7 @@
...
-# Arguments:
-# - REPO: Which repo is being released. `puppet-agent` or `puppet-agent-private`
-#         are the two primary options.
-# - WORKSPACE: Where the dependencies should be downloaded, e.g. `temp`.
-# - IGNORE_FOR: Which repos should be ignored (space-separated).
-# - OVERRIDE_PATH: A path to a txt file that will override versions.
-#
-# Deprecated arguments:
-# - PUPPET_AGENT_URL: This will still be read, but `REPO` is preferred.
+# See README.md for arguments and available environment variables.
```
Comment (caseywilliams) [inline pa_matchbatch.sh:5]: This was an incomplete list, so I replaced it with a reference to the README isntead of duplicating more

File: pa_matchbatch.sh [hunk 5]
```diff
@@ -207,42 +210,46 @@ for currentItem in ${repoRevMap}; do
 	to_rev=$(echo -n ${currentItem} | cut -d'|' -f1)
 	repo_url=$(echo -n ${currentItem} | cut -d'|' -f2)
 	repo=$(parseRepoName ${repo_url})
-	foss_name=$(stripPrivateSuffix ${repo})
+	public_name=$(stripPrivateSuffix ${repo})
```
Comment (caseywilliams) [inline pa_matchbatch.sh:213]: `foss_name` did not seem correct -- these are foss repos generally, just private forks of them (there are no enterprise features in the private forks, for example)

File: pa_matchbatch.sh [hunk 6]
```diff
@@ -207,42 +210,46 @@ for currentItem in ${repoRevMap}; do
...
+	  jiraProjectId=$(getJiraProjectIdFor "${public_name}")
+	  jiraFixedInProject=$(getJiraFixedInFor "${public_name}")
 
-	  versionsUsed="${versionsUsed}\n${foss_name}:${fix_ver}"
+	  versionsUsed="${versionsUsed}\n${public_name}:${fix_ver}"
 
-	  echo "Checking: ${foss_name}"
-	  echo "from_rev: $from_rev, to_rev: $to_rev, fix_ver: $fix_ver"
-	  ruby ${TICKETMATCH_PATH}/ticketmatch.rb --ci -f "${from_rev}" -t "${to_rev}" -p "${jiraProjectId}" -v "${jiraFixedInProject} ${fix_ver}" 2> /dev/null
+	  echo_bold "Ticketmatch results for $public_name"
```
Comment (caseywilliams) [inline pa_matchbatch.sh:243]: This more clearly separates the ticketmatch output sections by indenting the output for each component (and adds more human-friendly titles)

File: pa_matchbatch.sh [hunk 7]
```diff
@@ -207,42 +210,46 @@ for currentItem in ${repoRevMap}; do
...
 	popd
 done
 
-echo "<><><><><><><><><><>"
-echo "Versions used for JIRA searches (foss_repo:version)"
-printf "%b\n" ${versionsUsed}
+print_divider
+
+echo "The following versions were used for JIRA searches (repo_name:version)"
+echo_bold "If these versions are incorrect, you should create a version overrides file and try again. See the README."
```
Comment (caseywilliams) [inline pa_matchbatch.sh:254]: It's easy to miss incorrect automatically determined fixVersions in the output (this happened initially for 6.3.0), so I've put a bold note describing what to do if they're wrong here.

File: README.md [hunk 8]
```diff
@@ -75,37 +75,44 @@ Usage: ruby ticketmatch.rb [options]
...
+- Any circumstance where the reported version number is incorrect for some reason (a bad mergeup, for example)
 
-```export WORKSPACE="fully_qualified_path_to_a_dir"```
+To override the assumed version numbers for some or all components, create a text file with one
+`<repo-name>:<version>` entry on each line line. For example, this would override the assumed
+versions of puppet-agent and facter:
 
-invoke pa_matchbatch
+```
+puppet-agent:6.6.6
```
Comment (speedofdark) [inline README.md]: The agent of the beast?

File: README.md [hunk 9]
```diff
@@ -75,37 +75,45 @@ Usage: ruby ticketmatch.rb [options]
...
-```export WORKSPACE="fully_qualified_path_to_a_dir"```
+To override the assumed version numbers for some or all components, create a text file with one
+`<fixversion-prefix-for-jira-project>:<version>` entry on each line. For example, this would
+override the assumed versions of puppet-agent, facter, and pxp-agent:
 
-invoke pa_matchbatch
+```
+PUP:6.6.6
+FACT:3.13.0
+pxp-agent:1.11.0
```
Comment (caseywilliams) [inline README.md]: This format seems a little weird (using the jira fixVersion prefixes instead of the component names), but it seems like changing the behavior should be a separate PR.

File: pa_matchbatch.sh [hunk 10]
```diff
@@ -145,7 +146,7 @@ getJiraFixedInFor() {
 		;;
 		marionette-collective) echo MCO
 		;;
-		puppet-resource_api) echo PDK
+		puppet-resource_api) echo RSAPI
```
Comment (caseywilliams) [inline pa_matchbatch.sh:149]: The rsapi actually uses the `PDK` project but an `RSAPI <version>` fixVersion -- apparently fixVersions are not diligently applied, but it sounds like this is most correct

File: README.md (hunk 1 above)
Comment (speedofdark) [reply README.md:78]: (that's why we named it **pa_**matchbatch.sh :))

**Review by speedofdark — APPROVED** [review]

**Review by ScottGarman — APPROVED** [review]
